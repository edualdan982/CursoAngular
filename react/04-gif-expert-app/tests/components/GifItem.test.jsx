const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components");

describe("Prueba en <GifItem/>", () => {
  const titleTest = "Gif Item";
  const urlTest = "https://testing.prueba";

  test("debe hacer match con el snapshot", async () => {
    const { container } = render(<GifItem title={titleTest} url={urlTest} />);
    expect(container).toMatchSnapshot();
  });

  test('debe monstrar el titulo correctamente', () => {
    render(<GifItem title={titleTest} url={urlTest} />);

    expect(screen.getByText(titleTest)).toBeTruthy();
    // expect(screen.getAllByText(urlTest).length).toBeGreaterThan(0);
  });

  test('debe ir el url y alt correctament en la imagen', () => {
    render(<GifItem title={titleTest} url={urlTest} />);

    const {src, alt} = screen.getByRole( "img");
    expect(src).toContain(urlTest);
    expect(alt).toBe(titleTest);
  });
});
