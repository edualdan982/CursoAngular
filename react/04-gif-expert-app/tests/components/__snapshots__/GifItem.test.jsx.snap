// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif Item"
      src="https://testing.prueba"
    />
    <p>
      Gif Item
    </p>
  </div>
</div>
`;
